The job **`tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/19683090060/job/56381797360) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/56381797360) [(plain)](https://github.com/rust-lang/rust/commit/55b4137f5568ee8c541b12601c53c5468396f43b/checks/56381797360/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
linting javascript files and applying suggestions
Running eslint on rustdoc JS files

/checkout/src/librustdoc/html/static/js/search.js
  5322:5  error  Unexpected console statement  no-console

✖ 1 problem (1 error, 0 warnings)

rerun tidy with `--extra-checks=js:lint --bless` to apply eslint suggestions
tidy [extra_checks]: checks with external tool 'eslint' failed
tidy [extra_checks]: FAIL
tidy: The following check failed: extra_checks
Command `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools-bin/rust-tidy /checkout /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo /checkout/obj/build 4 /node/bin/yarn --extra-checks=py,cpp,js,spellcheck` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/tool.rs:1599:23
Executed at: src/bootstrap/src/core/build_steps/test.rs:1285:29

Command has failed. Rerun with -v to see more details.
Bootstrap failed while executing `test src/tools/tidy tidyselftest --extra-checks=py,cpp,js,spellcheck`
Build completed unsuccessfully in 0:02:46
  local time: Tue Nov 25 20:30:58 UTC 2025
  network time: Tue, 25 Nov 2025 20:30:58 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/src/librustdoc/html/static/js/search.js
+++ b/src/librustdoc/html/static/js/search.js
@@ -5319,8 +5319,6 @@ async function search(forced) {
     // recent search query is added to the browser history.
     updateSearchHistory(buildUrl(query.userQuery, filterCrates));
 
-    console.log(filterCrates);
-
     await showResults(
         docSearch,
         await docSearch.execQuery(query, filterCrates, window.currentCrate),
@@ -5401,14 +5399,19 @@ function registerSearchEvents() {
         }
         // up and down arrow select next/previous search result, or the
         // search box if we're already at the top.
+        //
+        // the .focus() calls are safe because there's no kind of element
+        // that lacks .focus() that should be in the document.
         if (e.which === 38) { // up
-            // @ts-expect-error
-            const previous = document.activeElement.previousElementSibling;
-            if (previous) {
-                // @ts-expect-error
-                previous.focus();
-            } else {
-                searchState.focus();
+            const active = document.activeElement;
+            if (active) {
+                const previous = active.previousElementSibling;
+                if (previous) {
+                    // @ts-expect-error
+                    previous.focus();
+                } else {
+                    searchState.focus();
+                }
             }
             e.preventDefault();
         } else if (e.which === 40) { // down

